build-system/tasks/runtime-test/runtime-test-base.js

```diff
@@ -126,8 +134,8 @@ function getFiles(testType) {
       if (argv.files) {
         return files.concat(getFilesFromArgv());
       }
-      if (argv.saucelabs) {
-        return files.concat(testConfig.unitTestOnSaucePaths);
+      if (argv.saucelabs || argv.safari || argv.firefox) {
+        return files.concat(testConfig.unitTestCrossBrowserPaths);
```

Comment:
I don't think it's safe to assume that having `--safari` or `--firefox` means this is running on GH actions.

Reply:
The old code equated cross browser with Sauce Labs, but this code does not equate cross browser with GH actions (hence the name `unitTestCrossBrowserPaths` and not `unitTestGithubActionsPaths`).

When unit tests are run on Safari / Firefox, only a subset of them are necessary and expected to work, and we therefore run only those.

SGTY?

Reply:
The problem is that this changes the files given to Karma.

Running `gulp unit --safari` on master uses these files:
```
[
  'test/_init_tests.js',
  'test-bin/test/fixtures/*.html',
  {
    pattern: 'test/fixtures/served/*.html',
    included: false,
    nocache: false,
    watched: true
  },
  {
    pattern: 'examples/**/*',
    included: false,
    nocache: false,
    watched: true
  },
  'test/unit/**/*.js',
  'ads/**/test/test-*.js',
  'ads/**/test/unit/test-*.js',
  'extensions/**/test/*.js',
  'extensions/**/test/unit/*.js'
]
```

Running `gulp unit --safari` on this branch use these:
```
[
  'test/_init_tests.js',
  'test-bin/test/fixtures/*.html',
  {
    pattern: 'test/fixtures/served/*.html',
    included: false,
    nocache: false,
    watched: true
  },
  {
    pattern: 'examples/**/*',
    included: false,
    nocache: false,
    watched: true
  },
  'test/unit/test-error.js'
]
```

Reply:
Aha, I see your point. The set of _all_ unit tests was never meant to work on Safari or Firefox, since they were primarily written to work on Chrome. When `--safari` or `--firefox` are used during local development, they are almost certainly accompanied by `--files`, which overrides this set in any case. (Today, running all unit tests on Safari of Firefox is guaranteed to result in several failures.)

Based on this, I think we should standardize the default set of tests run on Safari / Firefox to  a reduced set (`test-error.js` is just a seed value for now), and increase the number of tests in a separate PR.

Meanwhile, there is greater value in running all integration tests on multiple browsers, and that behavior remains unchanged.

WDYT?

/cc @choumx who had some opinions on this.

Reply:
+1 to Esther. What if I want to write a new unit test that should run cross-browser? Having to add it to `unitTestCrossBrowserPaths` first is not very intuitive.

I think we should decouple. Why not check `isGithubActionsBuild()`?

Reply:
I've decoupled the logic as requested. As a result, running `gulp unit --safari` on a local machine is guaranteed to throw up hundreds of failing tests (this is true today). If we want to address that, we can do so separately.

Thanks to both of you for the push back and for keeping me honest 😃